### PR TITLE
Add support for Global settings and Feature Flags

### PR DESCRIPTION
**Pull request by nwmac**
This PR adds a new global product 'Global Settings' with Advanced Settings and Feature Flags as per the Ember UI.

### CODE REVIEW

File: list/management.cattle.io.setting.vue [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  },
+
+  data() {
+    return { settings: null };
+  },
+
+  computed: {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    rows() {
```
Comment (richard-cox) [inline list/management.cattle.io.setting.vue:57]: Just for personal knowledge, would wrapping content with the Loading/fetch pending component mean `this.settings` could be used directly?

File: models/management.cattle.io.setting.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    let out = this._standardActions;
+
+    // Some settings are not editable
+    if (settingMetadata && settingMetadata.readOnly) {
+      toFilter.push('goToEdit');
+    }
+
+    out = out.filter((action) => {
+      if (!toFilter.includes(action.action)) {
+        return action;
```
Comment (richard-cox) [inline models/management.cattle.io.setting.js:18]: Just needs a `true`

File: config/settings.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  HIDE_LOCAL_CLUSTER:               'hide-local-cluster',
+  // CLUSTER_DEFAULTS:              'cluster-defaults',
+  AUTH_TOKEN_MAX_TTL_MINUTES:       'auth-token-max-ttl-minutes',
+  ENGINE_URL:                       'engine-install-url',
+  ENGINE_ISO_URL:                   'engine-iso-url',
+  FIRST_LOGIN:                      'first-login',
+  INGRESS_IP_DOMAIN:                'ingress-ip-domain',
+  PL:                               'ui-pl',
+  PL_RANCHER_VALUE:                 'rancher',
+  UI_BANNERS:                       'ui-banners',
```
Comment (richard-cox) [inline config/settings.js:36]: As an aside, should we hide this until we support it / show it as disabled / etc? Is there any open issue where we implement/update the underlying feature?

File: config/settings.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  UI_ISSUES:                        'ui-issues',
```
Comment (richard-cox) [inline config/settings.js:37]: There was some oddness with this. Editing the default blank value results in the ui showing it as a custom value. Assigned some junk value `dsgfgdf` and it fails to show this as `modified` in the list. Returning to edit the value and it's correctly marked as custom.

**Review by richard-cox — CHANGES_REQUESTED** [review]

File: assets/translations/en-us.yaml [hunk 5]
```diff
@@ -3205,3 +3225,62 @@ podAffinity:
...
   valuePlaceholder: e.g. bar
+
+##############################
+### Advanced Settings
+##############################
+
+advancedSettings:
+  header: Advanced Settings
+  subtext: "Typical users will not need to change these. Proceed with caution, incorrect values can break your {appName} installation. Settings which have been customized from default settings are tagged 'Modified'."
+  showLabel: "I understand that I can break things by changing advanced settings."
```
[the remark this answers is not in the document]
Comment (nwmac) [reply assets/translations/en-us.yaml:3236]: That's a straight port from the Ember UI, but it does not appear to be used, so I will remove

File: config/product/settings.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+    ifHaveType:          MANAGEMENT.SETTING,
+    inStore:             'management',
+    icon:                'globe',
+    weight:              -10,
+    removable:           false,
+    showClusterSwitcher: false,
+  });
+
+  virtualType({
+    label:          'Advanced Settings',
```
[the remark this answers is not in the document]
Comment (nwmac) [reply config/product/settings.js:31]: Done - added support for a labelKey for this. There are lots of other places we should use that in.

File: edit/management.cattle.io.setting.vue [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+          v-model="value.value"
+          :disabled="!customize"
+          :label="t('advancedSettings.edit.value')"
+        />
+      </div>
+    </div>
+  </CruResource>
+</template>
+
+<style lang="scss">
```
[the remark this answers is not in the document]
Comment (nwmac) [reply edit/management.cattle.io.setting.vue:152]: Done

File: list/management.cattle.io.feature.vue [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+            {{ t('generic.cancel') }}
+          </button>
+          <AsyncButton :mode="updateMode" class="btn bg-error ml-10" @click="toggleFlag" />
+        </template>
+      </Card>
+    </modal>
+  </div>
+</template>
+
+<style lang='scss'>
```
[the remark this answers is not in the document]
Comment (nwmac) [reply list/management.cattle.io.feature.vue:139]: Done

File: models/management.cattle.io.setting.js (hunk 2 above)
Comment (nwmac) [reply models/management.cattle.io.setting.js:18]: Simplified

File: config/settings.js (hunk 3 above)
Comment (nwmac) [reply config/settings.js:36]: Good question - I will file another issue - I guess it still covers the ember UI, but agreed we need to support this in the Vue UI

File: list/management.cattle.io.setting.vue (hunk 1 above)
Comment (nwmac) [reply list/management.cattle.io.setting.vue:57]: Seems to work that way

File: config/settings.js (hunk 4 above)
Comment (nwmac) [reply config/settings.js:37]: Yeah, fixed.

**Comment by nwmac** [comment]
@richard-cox I think I've addressed all of your feedback

**Review by richard-cox — APPROVED** [review]

**Comment by jiaqiluo** [comment]
https://github.com/rancher/dashboard/issues/2216

**Pull request merged**